File: GazeChim/ClientApp/src/app/models/bsff/bsff.ts
import { BsffDestination } from './bsff-destination';
import { BsffFicheIntervention } from './bsff-fiche-intervention';
import { BsffPackaging } from './packaging/bsff-packaging';
import { BsffWaste } from './bsff-waste';
import { BsffWeight } from './bsff-weight';
import BsffEmitter from './emission/bsff-emitter';
import { BsffType } from './enums/bsff-type';
import { BsffTransporter } from './transport/bsff-transporter';

export interface Bsff {
  id: string;
  createdAt: Date;
  updatedAt: Date;
  isDraft: boolean;
  type: BsffType;
  status: string;
  emitter: BsffEmitter;
  bsffPackagings: BsffPackaging[];
  waste: BsffWaste;
  weight: BsffWeight;
  transporter: BsffTransporter;
  destination: BsffDestination;
  ficheInterventions: BsffFicheIntervention[];
  forwarding: BsffPackaging[];
  grouping: BsffPackaging[];
  repackaging: BsffPackaging[];
}

File: GazeChim.Data/Models/Bsff.cs
using GazeChim.Data.Models.Enums;

namespace GazeChim.Data.Models
{
    public class Bsff : ITrackDechetModel
    {
        public string Id { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }
        public bool IsDraft { get; set; }
        public BsffType Type { get; set; }
        public string status { get; set; }
        public BsffEmitter Emitter { get; set; }
        public BsffPackaging[] BsffPackagings { get; set; }
        public BsffWaste Waste { get; set; }
        public BsffWeight Weight { get; set; }
        public BsffTransporter Transporter { get; set; }
        public BsffDestination Destination { get; set; }
        public BsffFicheIntervention[] FicheInterventions { get; set; }
        public BsffPackaging[] Forwarding { get; set; }
        public BsffPackaging[] Grouping { get; set; }
        public BsffPackaging[] Repackaging { get; set; }
    }
}
